homeassistant/components/lutron/__init__.py

```diff
@@ -36,36 +37,74 @@
 )
 
 
-def setup(hass, base_config):
+async def async_setup(hass, base_config):
     """Set up the Lutron component."""
-    hass.data[LUTRON_BUTTONS] = []
-    hass.data[LUTRON_CONTROLLER] = None
-    hass.data[LUTRON_DEVICES] = {
-        "light": [],
-        "cover": [],
-        "switch": [],
-        "scene": [],
-        "binary_sensor": [],
-    }
-
+    hass.data.setdefault(DOMAIN, {})
     config = base_config.get(DOMAIN)
-    hass.data[LUTRON_CONTROLLER] = Lutron(
+
+    if config is not None:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=config
+            )
+        )
+    return True
+
+
+class DummyArea:
+    """Fake area object.
+
+    This is used to tell all the lutron entities the names of the area they
+    belong to when we don't want to force the creation of those areas.
+    """
+
+    def __init__(self, name):
+        """Initialize the area."""
+        self.name = name
+        self.id = None
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a single Lutron deployment."""
+    if not entry.data:
+        return False
+
+    config = entry.data
+    device_registry = await dr.async_get_registry(hass)
+    area_registry = await ar.async_get_registry(hass)
+    hass_data = {
```

Comment:
Please name this something that's not so easily confused with `hass.data`.
```suggestion
    lutron_data = {
```